python-sdk/nuscenes/eval/panoptic/utils.py

```diff
@@ -0,0 +1,73 @@
+"""
+nuScenes-panoptic utils.
+Code written by Motional and the Robot Learning Lab, University of Freiburg.
+"""
+from typing import Dict
+
+from nuscenes.eval.lidarseg.utils import LidarsegClassMapper, get_samples_in_eval_set
+from nuscenes.nuscenes import NuScenes
+
+get_samples_in_panoptic_eval_set = get_samples_in_eval_set
+
+
+class PanopticClassMapper(LidarsegClassMapper):
+    """
+    Maps the (fine) classes to the (coarse) classes in the nuScenes-panoptic challenge.
+    Example usage::
+        nusc_ = NuScenes(version='v1.0-mini', dataroot='/data/sets/nuscenes', verbose=True)
+        mapper_ = PanopticClassMapper(nusc_)
+    """
+
+    def __init__(self, nusc: NuScenes):
+        """
+        Initialize a PanopticClassMapper object.
+        :param nusc: A NuScenes object.
+        """
+        super(PanopticClassMapper, self).__init__(nusc)
+        self.things = self.get_things()
+        self.stuff = self.get_stuff()
+
+    def get_stuff(self) -> Dict[str, int]:
+        """
+        Returns the mapping from the coarse class names to the coarse class indices for stuffs.
+        :return: A dictionary containing the mapping from the coarse class names to the coarse class indices for stuffs.
+            {
+              'driveable_surface': 11,
+              'other_flat': 12,
+              'sidewalk': 13,
+              'terrain': 14,
+              'manmade': 15,
+              'vegetation': 16
+            }
+        """
+        stuff_names = {'driveable_surface', 'other_flat', 'sidewalk', 'terrain', 'manmade', 'vegetation'}
+        coarse_name_to_id = self.get_coarse2idx()
+        assert stuff_names <= set(coarse_name_to_id.keys()), 'Invalid stuff names, pls check !'
+        stuff_name_to_id = {name: coarse_name_to_id[name] for name in stuff_names}
+
+        return stuff_name_to_id
+
+    def get_things(self) -> Dict[str, int]:
+        """
+        Returns the mapping from the coarse class names to the coarse class indices for things.
```

Comment:
What did things here refer to? Category?

Reply:
Can we use the same names everywhere? E.g. global/nuScenes and challenge classes?

Reply:
Addressed [181ac88](https://github.com/nutonomy/nuscenes-devkit/pull/628/commits/181ac8882dac5cb9df2fdf916622edf8ea46122f)